test/interfaces/background_field_interface.jl

```diff
@@ -0,0 +1,231 @@
+                groundtruth_envelope_values[idx] = _groundtruth_envelope(phi)
+            else
+                groundtruth_envelope_values[idx] = zero(phi)
+            end
+        end
+
+        @test isapprox(
+            test_envelope_values, groundtruth_envelope_values, atol=ATOL, rtol=RTOL
+        )
+    end
+end
+
+@testset "pulse amplitude" begin
+    @testset "$pol" for pol in (QEDfields.PolX(), QEDfields.PolY())
+        test_field = TestBGfield()
+
+        @testset "compute single" begin
+            rnd_phi = rand(RNG, RND_DOMAIN)
+            @test isapprox(
+                amplitude(test_field, pol, rnd_phi),
+                _groundtruth_amplitude(pol, rnd_phi),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, leftendpoint(RND_DOMAIN)),
+                _groundtruth_amplitude(pol, leftendpoint(RND_DOMAIN)),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, leftendpoint(RND_DOMAIN) - eps()),
+                zero(Float64),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, rightendpoint(RND_DOMAIN)),
+                _groundtruth_amplitude(pol, rightendpoint(RND_DOMAIN)),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, rightendpoint(RND_DOMAIN) + eps()),
+                zero(Float64),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+        end
+
+        @testset "compute vector" begin
+            rnd_phis = rand(RNG, RND_DOMAIN, 2)
+            push!(rnd_phis, leftendpoint(RND_DOMAIN))
+            push!(rnd_phis, leftendpoint(RND_DOMAIN) - eps())
+            push!(rnd_phis, rightendpoint(RND_DOMAIN))
+            push!(rnd_phis, rightendpoint(RND_DOMAIN) + eps())
+
+            test_amplitude_values = amplitude(test_field, pol, rnd_phis)
+
+            groundtruth_amplitude_values = Vector{Float64}(undef, length(rnd_phis))
+            for (idx, phi) in enumerate(rnd_phis)
+                if phi in RND_DOMAIN
+                    groundtruth_amplitude_values[idx] = _groundtruth_amplitude(pol, phi)
+                else
+                    groundtruth_amplitude_values[idx] = zero(phi)
+                end
+            end
+
+            @test isapprox(
+                test_amplitude_values, groundtruth_amplitude_values, atol=ATOL, rtol=RTOL
+            )
+        end
+    end
+end
+
+@testset "generic spectrum" begin
+    @testset "$pol" for pol in (QEDfields.PolX(), QEDfields.PolY())
+        test_field = TestBGfield()
+
+        @testset "compute single" begin
+            @testset "pnum = $rnd_pnum" for rnd_pnum in (
+                1 + (0.2 * rand(RNG) - 0.1), 1 - (0.2 * rand(RNG) - 0.1), 1.0, 0.0
```

Comment:
Isn't `0.2 * rand(RNG) - 0.1` the same as `0.1 * rand(RNG)`?

Also, `l_test` instead of `pnum` would be more in line with our common notation.

Reply:
It is not the same, because `0.2*rand(RNG)-0.1` can be negative, which `0.1*rand(RNG)` can't. However, thanks for pointing this out, because the intention here (and below as well) is, to have test values for `l` which is slightly below, and slightly above unity. Therefore, cb77bf0 changes this to `0.1*rand(RNG)`, because negative values make no sense at this point. 

Finally, cb77bf0 also changes `pnum` to `l_test`. However, at some point, we need to think about better naming of the photon-number parameter anyway.